### PR TITLE
feat: logging

### PR DESCRIPTION
**Pull request by joske**
Using println! for logging is annoying when snarkVM is used as library (like snarkOS).

- Replace println and friends with logging
- install simple_logger impl in binaries where appropriate

Draft, to be revisited after mainnet.

### CODE REVIEW

File: vm/package/run.rs [hunk 1]
```diff
@@ -36,7 +38,7 @@ impl<N: Network> Package<N> {
         let _locator = Locator::<N>::from_str(&format!("{program_id}/{function_name}"))?;
 
         #[cfg(feature = "aleo-cli")]
-        println!("🚀 Running '{}'...\n", _locator.to_string().bold());
+        trace!("🚀 Running '{}'...\n", _locator.to_string().bold());
```
Comment (howardwu) [inline vm/package/run.rs:41]: this is very wrong. the CLI does not use tracing!

File: circuit/types/boolean/src/helpers/subtractor.rs [hunk 2]
```diff
@@ -400,7 +404,8 @@ mod tests {
         check_true_sub_false_with_false(Mode::Public, Mode::Private, Mode::Constant, 0, 0, 2, 2);
         check_true_sub_false_with_true(Mode::Public, Mode::Private, Mode::Constant, 0, 0, 3, 3); // <- Differs
         check_true_sub_true_with_false(Mode::Public, Mode::Private, Mode::Constant, 0, 0, 2, 2);
-        check_true_sub_true_with_true(Mode::Public, Mode::Private, Mode::Constant, 0, 0, 3, 3); // <- Differs
+        check_true_sub_true_with_true(Mode::Public, Mode::Private, Mode::Constant, 0, 0, 3, 3);
+        // <- Differs
```
Comment (howardwu) [inline circuit/types/boolean/src/helpers/subtractor.rs:408]: incorrect changes here, please fix, thanks!

File: circuit/environment/src/helpers/linear_combination.rs [hunk 3]
```diff
@@ -105,21 +106,21 @@ impl<F: PrimeField> LinearCombination<F> {
             // Enforce property 2.
             // Note: This branch is triggered if ANY term is not (zero or one).
             if self.terms.iter().any(|(v, _)| !(v.value().is_zero() || v.value().is_one())) {
-                eprintln!("Property 2 of the `Boolean` type was violated in {self}");
+                warn!("Property 2 of the `Boolean` type was violated in {self}");
```
Comment (howardwu) [inline circuit/environment/src/helpers/linear_combination.rs:109]: these warnings are meant to be `eprintln`

File: algorithms/Cargo.toml [hunk 4]
```diff
@@ -156,6 +156,9 @@ optional = true
 [dependencies.num-traits]
 version = "0.2"
 
+[dependencies.log]
```
Comment (howardwu) [inline algorithms/Cargo.toml:159]: please remove usage of `log`

File: vm/package/run.rs (hunk 1 above)
Comment (howardwu) [reply vm/package/run.rs:41]: this has major downstream implications for `leo`

**Comment by joske** [comment]
@howardwu was this closed by accident?

**Pull request closed without merging**